### PR TITLE
Some fixes for OutputCoverallsJsonCoverage

### PR DESCRIPTION
**Pull request by markuspf**
* Check whether a file that occurs in the profiling data exists before
   outputting coverage (and trying to run md5sum on it)
 * Catch the case where pathtoremove is empty, because GAP's ReplacedString
   does not terminate in that case

### CODE REVIEW

File: gap/profiling.gi [hunk 1]
```diff
@@ -760,6 +761,14 @@ function(data, outfile, jobid, pathtoremove)
         return SplitString(str, " ")[1];
     end;
 
+    # GAP's ReplacedString does not terminate for
+    # empty string to replace
```
Comment (fingolfin) [inline gap/profiling.gi]: Probably should be a bug report for GAP, too? (perhaps you already made one, I did not check)

File: gap/profiling.gi [hunk 2]
```diff
@@ -783,13 +792,13 @@ function(data, outfile, jobid, pathtoremove)
     prev := false;
 
     for file in data.line_info do
-        if file[1] <> "stream" then
+        if IsExistingFile(file[1]) then
```
Comment (fingolfin) [inline gap/profiling.gi]: Could the same fix please be applied to the codecov generator function a few lines above this one?

File: gap/profiling.gi (hunk 1 above)
Comment (markuspf) [reply gap/profiling.gi]: I have indeed: https://github.com/gap-system/gap/issues/3134

File: gap/profiling.gi (hunk 2 above)
Comment (markuspf) [reply gap/profiling.gi]: Sure, though strictly speaking this is not necessary for codecov output (as we don't care whether the file currently exists).

I wonder whether we can compute the `md5sum` for a file and put that into the profiling output?

**Comment by ChrisJefferson** [comment]
Just checking, is this ready to be merged?

**Comment by markuspf** [comment]
Yeah, though I will make another PR that will add some more stuff (giving the ci service name, and outputting the option `parallel` in the json.

**Pull request merged**